### PR TITLE
Fix issues in gRPC tool

### PR DESCRIPTION
**Pull request by MadhukaHarith92**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/1773
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/1774
## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [x] Added tests

### CODE REVIEW

File: ballerina/tests/resources/proto-files/data-types/helloWorldWithNestedMessage.proto [hunk 1]
```diff
@@ -18,6 +18,8 @@
 
 syntax = "proto3";
 
+package choreo.apis;
```
Comment (daneshk) [inline ballerina/tests/resources/proto-files/data-types/helloWorldWithNestedMessage.proto:21]: shall we use a different package name. some random name

Comment (MadhukaHarith92) [reply ballerina/tests/resources/proto-files/data-types/helloWorldWithNestedMessage.proto]: Updated in 527e805

File: native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/SyntaxTreeGenerator.java [hunk 2]
```diff
@@ -111,14 +109,33 @@ public static SyntaxTree generateSyntaxTree(StubFile stubFile, boolean isRoot) {
...
+        for (ServiceStub serviceStub : stubFile.getStubList()) {
+            List<Method> methodList = new ArrayList<>();
+            methodList.addAll(serviceStub.getUnaryFunctions());
+            methodList.addAll(serviceStub.getClientStreamingFunctions());
+            methodList.addAll(serviceStub.getServerStreamingFunctions());
+            methodList.addAll(serviceStub.getBidiStreamingFunctions());
+
+            if (checkForImportsInServices(methodList, "time:Utc")
+                    || checkForImportsInServices(methodList, "time:Seconds")) {
                 ImportDeclarationNode importForTime = Imports.getImportDeclarationNode(
```
Comment (dilanSachi) [inline native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/SyntaxTreeGenerator.java:123]: I think we can remove this

Comment (MadhukaHarith92) [reply native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/SyntaxTreeGenerator.java]: Updated in 6d5ba05

File: native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/SyntaxTreeGenerator.java [hunk 3]
```diff
@@ -111,14 +109,30 @@ public static SyntaxTree generateSyntaxTree(StubFile stubFile, boolean isRoot) {
...
+        }
+        boolean importTime = false;
+        for (ServiceStub serviceStub : stubFile.getStubList()) {
+            List<Method> methodList = new ArrayList<>();
+            methodList.addAll(serviceStub.getUnaryFunctions());
+            methodList.addAll(serviceStub.getClientStreamingFunctions());
+            methodList.addAll(serviceStub.getServerStreamingFunctions());
+            methodList.addAll(serviceStub.getBidiStreamingFunctions());
+
+            if (checkForImportsInServices(methodList, "time:Utc")
```
Comment (dilanSachi) [inline native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/SyntaxTreeGenerator.java:121]:
```suggestion
            if (!importTime && (checkForImportsInServices(methodList, "time:Utc")
```
Maybe we can avoid redundant checks in this way

File: native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/SyntaxTreeGenerator.java [hunk 4]
```diff
@@ -111,14 +109,30 @@ public static SyntaxTree generateSyntaxTree(StubFile stubFile, boolean isRoot) {
...
         }
+        if (checkForImportsInMessageMap(stubFile, "Timestamp") ||
+                checkForImportsInMessageMap(stubFile, "Duration")) {
+            importTime = true;
+        }
+        if (importTime) {
+            ImportDeclarationNode importForTime = Imports.getImportDeclarationNode(
+                    "ballerina", "time"
+            );
+            imports = imports.add(importForTime);
```
Comment (daneshk) [inline native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/SyntaxTreeGenerator.java:134]: @MadhukaHarith92 is this change adding an import to the correct generated file. there can be scenarios where we need to add the import to both the root file and dependent file.

Comment (MadhukaHarith92) [reply native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/SyntaxTreeGenerator.java:134]: Yes. Added a test case where the import gets added to both root and dependent files in 7db7f64

File: ballerina/tests/resources/proto-files/data-types/time_dependent.proto [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ */
+
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
+
+message Greeting {
+	string name = 1;
+	google.protobuf.Timestamp time = 2;
+}
```
Comment (tharindulak) [inline ballerina/tests/resources/proto-files/data-types/time_dependent.proto:26]: Shall we add a new line here ?

Comment (MadhukaHarith92) [reply ballerina/tests/resources/proto-files/data-types/time_dependent.proto]: Added in 5b8a515

File: native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/SyntaxTreeGenerator.java (hunk 3 above)
Comment (MadhukaHarith92) [reply native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/SyntaxTreeGenerator.java:121]: Added a break in 5b8a515

**Review by daneshk — APPROVED** [review]

**Review by dilanSachi — APPROVED** [review]

**Pull request merged**